modules/economy-system/commands/shop.js

```diff
@@ -1,92 +1,42 @@
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: interaction.user.tag,
-            i: item['value']
-        }));
+        if (!checkPerms(interaction)) return;
+        await createShopItem(interaction);
+        shopMsg(interaction.client);
     },
     'buy': async function (interaction) {
-        const itemName = await interaction.options.get('item');
-        const item = await interaction.client.models['economy-system']['Shop'].findOne({
-            where: {
-                name: itemName['value']
-            }
-        });
-        if (!item) {
-            interaction.reply({
-                content: interaction.client.configurations['economy-system']['strings']['notFound'],
-                ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
-            });
-        }
-        const user = await interaction.client.models['economy-system']['Balance'].findOne({
-            where: {
-                id: interaction.user.id
-            }
-        });
-        if (user.balance < item.price) return interaction.reply({
-            content: interaction.client.configurations['economy-system']['strings']['notEnoughMoney'],
-            ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
-        });
-        await editBalance(interaction.client, interaction.user.id, 'remove', item.price);
-        await interaction.member.roles.add(item.role);
-        createleaderboard(interaction.client);
-        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['buyMsg'], {'%item%': itemName['value']}, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        interaction.client.logger.info(`[economy-system] ` + localize('economy-system', 'user-purchase', {
-            u: interaction.user.tag,
-            i: item['value'],
-            p: item['price']
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'user-purchase', {
-            u: interaction.user.tag,
-            i: item['value'],
-            p: item['price']
-        }));
+        let name = await interaction.options.get('item-name');
+        if (name) name = name['value'];
+        let id = await interaction.options.get('item-id')['value'];
+        if (id) id = id['value'];
+        buyShopItem(interaction, id, name);
     },
     'list': async function (interaction) {
-        const msg = await createShopMsg(interaction.client);
+        const msg = await createShopMsg(interaction.client, interaction.guild, !interaction.client.configurations['economy-system']['config']['publicCommandReplies']);
         interaction.reply(msg);
     },
     'delete': async function (interaction) {
```

Comment:
This command will cause crash if no parameters were set

Reply:
This command doesn't have any parameters. Or what do you mean by "if no parameters were set"?

Reply:
I think I selected something wrong here… What I meant by that is a crash if users try to /shop buy an item, but do not set any parameters in the command (I think, but I'm not sure myself so…)